app/controllers/dwellings_controller.rb

```diff
@@ -6,14 +6,18 @@ def index
     @dwelling = Dwelling.new
   end
 
+  def new
+    @dwelling = Dwelling.new
+  end
+
   def create
     @dwelling = Dwelling.new(dwelling_params.merge(development: @development))
     if @dwelling.save
       flash[:notice] = 'Dwelling successfully added'
       redirect_to development_dwellings_path(@development)
     else
       @dwellings = @development.dwellings
```

Comment:
Because we're no longer rendering the index page on error, we don't need to load the dwellings list, so can remove this line.